### ISSUE TITLE
Incompatible with KSP and Kotlin 1.9.0

### ISSUE THREAD

**Issue by dalewking** [icerockdev/moko-resources#535]
Have a KMM project that uses Moko Resources and KSP. When I upgraded to Kotlin 1.9.0 gradle fails immediately with an error:

> Cannot change attributes of dependency configuration ':myProject:iosX64CInteropApiElements' after it has been resolved

Doing some debugging I find that I don't get the error in another module that doesn't use moko resources and if I disable moko resources in the failing project the error goes away.

**Comment by glockbender** [comment]
Trying to build the latest moko-resources with Kotlin 1.9.0 and receiving build error on `gradlew build`
![image](https://github.com/icerockdev/moko-resources/assets/16047417/2848baa5-24d8-4465-8a38-b7bef21b3193)
`Reason: Task ':compileKotlinJvm' uses this output of task ':generateMRcommonMain' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.`

**Comment by Ric-AB** [comment]
I'm also facing this issue : (

**Comment by luca992** [comment]
I think I've figured out that it will not have this issue if:

```kotlin
kotlin {
    jvm()
    sourceSets {
       val commonMain by getting
        // jvmMain needs to be set to explicity dependsOn(commonMain) even though it should be implied
        val jvmMain by getting {
            dependsOn(commonMain)
        }
    }
}
```

**Comment by dalewking** [comment]
> I think I've figured out that it will not have this issue if:
> 
> ```kotlin
> kotlin {
>     jvm()
>     sourceSets {
>        val commonMain by getting
>         // jvmMain needs to be set to explicity dependsOn(commonMain) even though it should be implied
>         val jvmMain by getting {
>             dependsOn(commonMain)
>         }
>     }
> }
> ```

That has no bearing on the original subject of this issue

**Comment by luca992** [comment]
> > I think I've figured out that it will not have this issue if:
> > 
> > ```kotlin
> > kotlin {
> >     jvm()
> >     sourceSets {
> >        val commonMain by getting
> >         // jvmMain needs to be set to explicity dependsOn(commonMain) even though it should be implied
> >         val jvmMain by getting {
> >             dependsOn(commonMain)
> >         }
> >     }
> > }
> > ```
> 
> That has no bearing on the original subject of this issue

Ah. Sorry, I was replying to @glockbender .

**Comment by prof18** [comment]
Is there a rough ETA for resolving this issue?

**Comment by Lilytreasure** [comment]
I am facing similar issue too

**Comment by Moozart** [comment]
Same issue on kotlin 1.9.0.   +1

**Comment by ldobransky** [comment]
same issue on my side.

**Comment by alaegin** [comment]
I'm struggling with the same issue, unfortunately

**Comment by prof18** [comment]
YOu can circumvent the issue by having the translations in a module that doesn't use KSP. 
For example: https://github.com/prof18/feed-flow/tree/main/i18n

**Comment by eric-ampire** [comment]
it's not possible to be able to download the version 0.23.1 on a separate branch ?
Since moko is strongly tied to the version of kotlin, while the stable version is only compatible with Kotlin 1.8, it should be great to be able to download a snapshot version of moko that is compatible with the latest version of Kotlin

**Comment by Lawrance3** [comment]
I have the same issue. My project heavily relies on moko-resources, so it's crucial for me to resolve this. Does anybody have a workaround or some insights into this issue? It would be immensely helpful!

**Comment by prof18** [comment]
> I have the same issue. My project heavily relies on moko-resources, so it's crucial for me to resolve this. Does anybody have a workaround or some insights into this issue? It would be immensely helpful!

@Lawrance3 As I mentioned above you can create a module that only uses moko resources and no KSP. In this way everything works

https://github.com/icerockdev/moko-resources/issues/535#issuecomment-1727622216

**Comment by Breens-Mbaka** [comment]
my project doesn't use KSP and I'm facing similar issue

"Cannot change attributes of dependency configuration ':shared:iosX64CInteropApiElements' after it has been resolved"

For some reason I had the plugin added successfully for the 1st time. then removed it. Then after a while I tried adding the plugin and it brought up the above issue.

My Kotlin version is 1.9.20 and gradle version is 8.1.0-beta02
![Uploading Screenshot 2023-12-05 at 10.26.53.png…]()
